broadinstitute/scp-ingest-pipeline
# Extending differential expression analysis to AnnData files (SCP-5764)

#### BACKGROUND & CHANGES
This update adds `h5ad` as a valid `matrix_file_type` for executing differential expression analyses.  Our current differential expression implementation in Scanpy uses a temporary h5ad file created from source plain text files, so allowing the use of user-uploaded AnnData files is very simple.  Aside from updating the CLI, the only changes are to use the `raw` slot from the matrix and to not transpose the data.

#### MANUAL TESTING
1. Initialize your local dev environment as usual
2. Run the example command from `ingest_pipeline.py` for AnnData DE and confirm you see similar output:
```
python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 differential_expression --annotation-name louvain --annotation-type group --annotation-scope study --matrix-file-path ../tests/data/anndata/trimmed_compliant_pbmc3K.h5ad --matrix-file-type h5ad --annotation-file ../tests/data/anndata/h5ad_frag.metadata.tsv --cluster-file ../tests/data/anndata/h5ad_frag.cluster.X_umap.tsv --cluster-name umap --study-accession SCPdev --differential-expression
...
STATUS after DE [0]
distinct_id: 2f30ec50-a04d-4d43-8fd1-b136a2045079
studyAccession: SCPdev
fileName: dec0dedfeed1111111111111
fileType: input_validation_bypassed
fileSize: 1
trigger: dev-mode
logger: ingest-pipeline
appId: single-cell-portal
file_path ../tests/data/anndata/h5ad_frag.cluster.X_umap.tsv
study_file_id dec0dedfeed1111111111111
```
3. Confirm you do not get an error, and then open the `umap--louvain--0--study--wilcoxon.tsv` file and confirm the first 3 entries are as follows:
```
	names	scores	logfoldchanges	pvals	pvals_adj	pct_nz_group	pct_nz_reference
0	LDHB	29.36	2.091	1.776e-189	2.435e-185	0.9196	0.5019
1	RPS27	27.56	1.005	3.229e-167	1.217e-163	0.9991	0.9896
2	RPS25	26.45	1.076	3.659e-154	7.168e-151	0.9982	0.9689
...
```
